editor/plugins/sprite_frames_editor_plugin.cpp

```diff
@@ -949,13 +949,16 @@ void SpriteFramesEditor::_animation_name_edited() {
 	String name = new_name;
 	int counter = 0;
 	while (frames->has_animation(name)) {
+		if (name == String(edited_anim)) {
+			edited->set_text(0, name); // The name didn't change, just updated the column text to name.
+			return;
+		}
 		counter++;
 		name = new_name + "_" + itos(counter);
 	}
 
 	EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
 	undo_redo->create_action(TTR("Rename Animation"), UndoRedo::MERGE_DISABLE, EditorNode::get_singleton()->get_edited_scene());
-	_rename_node_animation(undo_redo, false, edited_anim, "", "");
```

Comment:
To be honest, I don't really remember what this initialization was implemented for. The `_rename_node_animation()` is a method to change the animation name kept in other properties of AnimatedSprite, but without it, it is possible to have some inconsistencies in some properties, so it should be tested carefully.

Reply:
The previous code will rename `name` to a non-empty value. Even though it would have side effects, I feel like it could be done in the call on line 961.

https://github.com/godotengine/godot/blob/57919beb05687cf2ee80ed0d3ccac6445c7edea9/editor/plugins/sprite_frames_editor_plugin.cpp#L958-L962